src/java.base/share/classes/java/util/stream/Collectors.java

```diff
@@ -1972,6 +1972,35 @@ R get() {
             this.forFalse = forFalse;
         }
 
+        @Override
+        public int size() {
+            return 2;
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return false;
+        }
+
+        @Override
+        public T get(Object key) {
+            if (key instanceof Boolean b) {
+                return b.booleanValue() ? forTrue : forFalse;
```

Comment:
Can we drop `.booleanValue()` ?
```suggestion
                return b ? forTrue : forFalse;
```